### PR TITLE
CEDS-1047 Implement Standard/FullDec journey

### PR DESCRIPTION
**Pull request by PauloPortugal**
* We have added a ActionRefiner as an action composition enhancement in
order to allows us to add context and processing to the HTTP request,
by providing which jorney type the user is in.

* This results in an extra DB call, which although it may seem unnecessary,
provides the ability in the future to resume the journey and knowing
which journey the user is in.

* At this stage we have decided to implement in one one controller
    (`RepresentativeDetailsPageController`) as a proof of concept and
    because we already have one story ready to be played that affects that
    same controller.
    
 
* In order to allow Standard and Supplementary user journeys we had to
remove the `supplementary` and `standard` from the URI, because
duplicating the routes files will make Play always map the controller to
the last URI found on the `app.routes` file.

* Doing so meant that the unit tests had to be updated as well.

### CODE REVIEW

File: app/controllers/ChoiceController.scala [hunk 1]
```diff
@@ -60,8 +60,7 @@ class ChoiceController @Inject()(
               case SupplementaryDec =>
                 Redirect(controllers.supplementary.routes.DeclarationTypeController.displayDispatchLocationPage())
               case StandardDec =>
-                // TODO Standard declaration is not supported now
-                Redirect(controllers.routes.ChoiceController.displayChoiceForm())
+                Redirect(controllers.supplementary.routes.DeclarationTypeController.displayDispatchLocationPage())
```
[the remark this answers is not in the document]
Comment (PauloPortugal) [reply app/controllers/ChoiceController.scala]: :man_facepalming:

File: conf/app.routes [hunk 2]
```diff
@@ -1,7 +1,8 @@
 
 ->         /movement                                    movement.Routes
 
-->         /declaration/supplementary                   supplementary.Routes
+->         /declaration                                 supplementary.Routes
```
[the remark this answers is not in the document]
Comment (PauloPortugal) [reply conf/app.routes]: This is the thing that breaks everything.

**Comment by PauloPortugal** [comment]
@PatrykRudnicki I have addressed all of your comments. I have also amended your fix and included your effort in the commit message. Many thanks for the help!!! :)

**Comment by PatrykRudnicki** [comment]
Just question about this `Conflict` as we discussed :)

**Pull request merged**